File: BufferConsole/IConsoleOutput.cs
using System;

namespace Omnicatz.Console
{
    public interface IConsoleOutput
    {
        void Flush();
        Cell[,] Buffer { get; }
        void Clear();
        void Print(char c);     
        void ResetColor();
        void Reset();
        void BreakRow();
        int CursorX { get; set; }
        int CursorY { get; set; }
        int Indent { get; set; }
        ConsoleColor Fore { get; set; }
        ConsoleColor Back { get; set; }
        void Print(string text);        
        }
}

File: TSConsle/wwwroot/Console/IConsoleOutput.ts
import { Cell } from "./Cell.js";
import { Color } from "./Color.js";

export interface IConsoleOutput {
    GetBuffer(): Array<Array<Cell>>;
    Clear(): void;
    PrintChar(c: string): void;
    ResetColor(): void;
    Reset(): void;
    BreakRow(): void;
    CursorX: number;
    CursorY: number;
    Indent: number;
    Fore: Color;
    Back: Color;
    Print(text: string): void;        
}